### DISCUSSION TITLE
Where to find list of usable components list to create a new device type

### DISCUSSION THREAD

**Discussion by teomcam** [netbox-community/devicetype-library#2069] [Q&A] [answered]
I need to create a new device type for security equipment. Before I start creating the template file, I want to use the most accurate port definitions possible. Is there any documentation available that lists all possible ports and components?

**Comment by danner26** [comment, marked as answer]
Hello, the [README](https://github.com/netbox-community/devicetype-library/blob/master/README.md) has the most up to date information regarding how to structure a definition file. You can also look within the [schema folder](https://github.com/netbox-community/devicetype-library/tree/master/schema) on this repo for specific information regarding what port types are available and such.

For instance, if you wanted to see what type of interfaces we currently accept, you could navigate to the [devicetype schema](https://github.com/netbox-community/devicetype-library/blob/master/schema/devicetype.json) which when reading through will lead you to the [generated_schema](https://github.com/netbox-community/devicetype-library/blob/master/schema/generated_schema.json). Within the `generated_schema.yml` you will find `interface` which has a whole list of interface types.

You can also look at any of the [recently approved and merged PR's](https://github.com/netbox-community/devicetype-library/pulls?q=is%3Apr+is%3Aclosed+review%3Aapproved) which can give you a good idea on accepted formats.

If you have anymore questions, please feel free to let us know!

**Reply by teomcam** [reply]
Legend, thank you so much
